# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1297729691 -3600
#      Tue Feb 15 01:28:11 2011 +0100
# Node ID 890e89c3dfebad34de2d6a8e36e79f23f35b2440
# Parent  f899f451fe76b9d5a00219f4ecc1ce71801f08fa
More explicit error message when source with tests for dynamically linked functions not found (#30341)

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,13 @@
 2011-02-14  David Bateman  <dbateman@free.fr>
 
+	* testfun/test.m: More explicit error message when source of a
+	dynamically linked function is not found.
+
 	* plot/private/__scatter__.m: Correctly set the facevertexcdata and
 	faces properties in the patch objects.
 
 2011-02-14  Marco Caliari <marco.caliari@univr.it>
 
 	* polynomial/spline.m: Allow length(x) == 2 and unsorted x values.
 
 2011-02-13  Konstantinos Poulios  <logari81@gmail.com>
diff --git a/scripts/testfun/test.m b/scripts/testfun/test.m
--- a/scripts/testfun/test.m
+++ b/scripts/testfun/test.m
@@ -168,17 +168,21 @@ function [__ret1, __ret2, __ret3, __ret4
       __file = __file{1};
     endif
   endif
   if (isempty (__file))
     if (__grabdemo)
       __ret1 = "";
       __ret2 = [];
     else
-      fprintf (__fid, "%s%s does not exist in path\n", __signal_empty, __name);
+      if (exist (__name) == 3)
+        fprintf (__fid, "%s%s source code with tests for dynamically linked function not found\n", __signal_empty, __name);
+      else
+        fprintf (__fid, "%s%s does not exist in path\n", __signal_empty, __name);
+      endif
       fflush (__fid);
       if (nargout > 0)
         __ret1 = __ret2 = 0;
       endif
     endif
     if (__close_fid)
       fclose(__fid);
     endif
